base_layer/core/src/covenants/arguments.rs

```diff
@@ -183,6 +187,15 @@ impl CovenantArg {
 macro_rules! require_x_impl {
     ($name:ident, $output:ident, $expected: expr, $output_type:ident) => {
         #[allow(dead_code)]
+        /// `require_x_impl!` is a helper macro that generates an implementation of a function with a specific signature
+        /// based on the provided input parameters. Functionality:
+        /// The macro expects to receive either three or four arguments.
+        ///     $name, represents the name of the function to be generated.
+        ///     $output, represents the name of the enum variant that the function will match against.
+        ///     $expected, represents an expression that will be used in the error message when the provided argument
+        ///         does not match the expected variant.
+        ///     (optional) $output_type, represents the type that the function will return. If
+        ///         not provided, it defaults to the same as $output.
```

Comment:
This documents the macro, not the function the macro makes - so this should go above 
macro_rules! require_x_impl {